tools/wavetool.py

```diff
@@ -0,0 +1,279 @@
+#! /usr/bin/python3
+
+import os
+import sys
+import argparse
+import numpy as np
+import scipy.signal as signal
+import scipy.io.wavfile as wavefile
+import scipy.fftpack as fftpack
+
+eps = 0.0000000001 # used to avoid zero division error
+
+class WavGenerator:
+    @staticmethod
+    def generateSine(param_dict):
+        time = np.arange(0, float(param_dict['duration']), 1.0/float(param_dict['sample_rate']), dtype=np.float32)
+        data = float(param_dict['amp']) * np.sin(2 * np.pi * float(param_dict['freq']) * time + float(param_dict['phase']))
+        return np.reshape(np.repeat(data, int(param_dict['channel'])),[len(data), int(param_dict['channel'])])
+
+    @staticmethod
+    def generateCosine(param_dict):
+        time = np.arange(0, float(param_dict['duration']), 1.0/float(param_dict['sample_rate']), dtype=np.float32)
+        data = float(param_dict['amp']) * np.cos(2 * np.pi * float(param_dict['freq']) * time + float(param_dict['phase']))
+        return np.reshape(np.repeat(data, int(param_dict['channel'])),[len(data), int(param_dict['channel'])])
+
```

Comment:
These functions are nearly identical to each other which makes it very difficult to see the differences between them. Make "sin" and "cos" an  "f" parameter and you can have only one function, no duplication, less code and less maintenance.